src/main/java/io/swagger/codegen/v3/generators/examples/ExampleGenerator.java

```diff
@@ -269,10 +272,10 @@ private double randomNumber(Double min, Double max) {
     }
 
     private Object resolveModelToExample(String name, String mediaType, Schema schema, Set<String> processedModels) {
-        if (processedModels.contains(name)) {
+        if (processedModels.contains(schema.get$ref())) {
             return schema.getExample();
         }
-        processedModels.add(name);
+        processedModels.add(schema.get$ref());
```

Comment:
These `processedModels` are passed to `resolveSchemaToExample()` at line 290.

I changed this from using the `name` because names are not guaranteed to be unique within an object tree. For example, an `id` property on the top-level schema could be redundant with an `id` in a nested object. In the case of a self-referencing or circular structure, names are certain _not_ to be unique.